pennylane/devices/default_tensor.py

```diff
@@ -258,6 +258,10 @@ def circuit(num_qubits):
 
     We can provide additional keyword arguments to the device to customize the simulation. These are passed to the ``quimb`` backend.
 
+    .. note::
+        Be aware that `quimb` uses multi-threading with `numba <https://numba.pydata.org/numba-doc/dev/user/threading-layer.html>`_ as well as for linear algebra operations with `numpy.linalg <https://numpy.org/doc/stable/reference/routines.linalg.html#linear-algebra-numpy-linalg>`_. Proper setting of the corresponding environment variables (e.g. `OMP_NUM_THREADS`, `OPENBLAS_NUM_THREADS`, `NUMBA_NUM_THREADS` etc.) depending on your hardware is highly recommended and will have a strong impact on the device's performance.
+        To avoid a slowdown in performance for circuits with more than 10 wires. We recommend you to set the environment variable `OPENBLAS_NUM_THREADS=1` or `MKL_NUM_THREADS=1`, depending on your NumPy packages & accelerated linear algebra libraries or using  `threadpoolctl <https://github.com/joblib/threadpoolctl>`_ package.
```

Comment:
```suggestion
        To avoid a slowdown in performance for circuits with more than 10 wires, we recommend setting the environment variable relevant for your BLAS library backend (e.g. `OMP_NUM_THREADS=1`, `OPENBLAS_NUM_THREADS=1` or `MKL_NUM_THREADS=1`), depending on your NumPy package & associated libraries. Alternatively, you can use  `threadpoolctl <https://github.com/joblib/threadpoolctl>`_  to limit the threads within your executing script. For optimal performance you can adjust the number of threads to find the best fit for your workload.
```

I think we need to give them more context on tuning here, since this will boil to a lot of BLAS-3 calls under the hood. As soon as we tell people to use `1` they may assume that is the best, and never adjust --- taking the sample workload with a much deeper circuit and going 1,2,4... showed me pretty good performance gains.